hoot-core/src/main/cpp/hoot/core/io/SqlBulkDelete.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2016, 2017 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "SqlBulkDelete.h"
+
+// hoot
+#include <hoot/core/util/HootException.h>
+#include <hoot/core/util/Log.h>
+
+// Qt
+#include <QtSql/QSqlDatabase>
+#include <QSqlError>
+#include <QStringBuilder>
+
+// tgs
+#include <tgs/System/Time.h>
+
+namespace hoot
+{
+
+SqlBulkDelete::SqlBulkDelete(QSqlDatabase& db, const QString tableName) :
+_db(db),
+_tableName(tableName)
+{
+  _time = 0;
+}
+
+SqlBulkDelete::~SqlBulkDelete()
+{
+  LOG_DEBUG("(" << _tableName << ") Total time inserting: " << _time);
```

Comment:
I think you mean "deleting" instead of "inserting". Also "pending deletes" below, instead of "pending inserts"

Reply:
definitely.  will fix.